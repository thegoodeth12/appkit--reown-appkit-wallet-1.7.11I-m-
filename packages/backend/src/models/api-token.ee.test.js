import { describe, it, expect, vi } from 'vitest';
import ApiToken from '@/models/api-token.ee.js';

describe('ApiToken model', () => {
  it('tableName should return correct name', () => {
    expect(ApiToken.tableName).toBe('api_tokens');
  });

  it('jsonSchema should have correct validations', () => {
    expect(ApiToken.jsonSchema).toMatchSnapshot();
  });

  describe('assignToken', () => {
    it('should assign a new token', async () => {
      const apiToken = new ApiToken();
      await apiToken.assignToken();

      expect(apiToken.token).toBeDefined();
    });
  });

  describe('beforeInsert', () => {
    it('should call assignToken method', async () => {
      const apiToken = new ApiToken();
      const assignTokenSpy = vi.spyOn(apiToken, 'assignToken');

      await apiToken.$beforeInsert();
      expect(assignTokenSpy).toHaveBeenCalled();
    });
  });
});
