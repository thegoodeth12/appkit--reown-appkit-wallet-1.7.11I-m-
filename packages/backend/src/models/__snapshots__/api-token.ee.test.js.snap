// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ApiToken model > jsonSchema should have correct validations 1`] = `
{
  "properties": {
    "createdAt": {
      "type": "string",
    },
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "token": {
      "minLength": 32,
      "type": "string",
    },
    "updatedAt": {
      "type": "string",
    },
  },
  "type": "object",
}
`;
